app/src/main/java/com/berniesanders/connect/BaseActivity.java

```diff
@@ -0,0 +1,59 @@
+package com.berniesanders.connect;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+import com.berniesanders.connect.hook.ActivityHook;
+
+public class BaseActivity extends AppCompatActivity {
+    private Iterable<ActivityHook> mHooks;
+
+    protected void setHooks(final Iterable<ActivityHook> hooks) {
+        mHooks = hooks;
+    }
+
+    @Override
+    public void onCreate(final Bundle savedInstanceState) {
```

Comment:
im a fan of keeping this protected in the base class, not that it really matters